mintbean-calc/src/App.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from "react";
+import "./App.css";
+
+class App extends Component {
+  constructor(){
+    super()
+    this.state = {
+      payment: null,
+      amount: null,
+      inputRate: null,
+      rate: null,
+      periods: null
+    }
+  }
+  handleChange = e => {
+    if(e.target.name === "inputRate"){
+      this.setState({
+         rate: e.target.value/100/12
+      })
+    }
+    this.setState({
+      [e.target.name]: parseInt(e.target.value)
+    })
+  }
+  componentDidUpdate = e => {
+    
+  }
+  calculate = e => {
+    let p = this.state.amount
```

Comment:
You can probably use the spread operator here. let [payment,amount,rate...] = this.state; Not 100% sure on teh syntax

Reply:
I probably should have done this:
calculate = () => {
}
using an arrow function for the component methods binds the method to that function.  You could alternatively use a this.bind method by putting this:
```
this.calculate = this.calculate.bind(this)
```
at the top
and then use
```
calculate() {
}
```